webauthn4j-core/src/test/java/com/webauthn4j/data/client/OriginTest.java

```diff
@@ -49,16 +50,86 @@ void equals_test() {
         );
     }
 
+    @Test
+    void apk_key_hash_equals_test() {
+        Origin apkKeyHashSha1_1 = new Origin("android:apk-key-hash:pNiP5iKyQ8JwgGOaKA1zGPUPJIS-0H1xKCQcfIoGLck");
+        Origin apkKeyHashSha1_2 = new Origin("android:apk-key-hash","pNiP5iKyQ8JwgGOaKA1zGPUPJIS-0H1xKCQcfIoGLck");
+        Origin apkKeyHashSha256_1 = new Origin("android:apk-key-hash-sha256:xT5ZucZJ9N7oq3j3awG8J/NlKf8trfo6AAJB8deuuNo=");
+        Origin https_examplecom_default = new Origin("https://example.com");
+
+        assertAll(
+                () -> assertThat(apkKeyHashSha1_1).isEqualTo(apkKeyHashSha1_2),
+                () -> assertThat(apkKeyHashSha1_1).isNotEqualTo(https_examplecom_default),
+                () -> assertThat(apkKeyHashSha1_1).isNotEqualTo(apkKeyHashSha256_1),
+                () -> assertThat(apkKeyHashSha1_2).isNotEqualTo(https_examplecom_default)
+        );
+    }
+
+    @Test
+    void apk_key_hash_sha256_equals_test() {
+        Origin apkCertHashSha256_1 = new Origin("android:apk-key-hash-sha256:xT5ZucZJ9N7oq3j3awG8J/NlKf8trfo6AAJB8deuuNo=");
+        Origin apkCertHashSha256_2 = new Origin("android:apk-key-hash-sha256","xT5ZucZJ9N7oq3j3awG8J/NlKf8trfo6AAJB8deuuNo=");
+        Origin apkCertHashSha1_1 = new Origin("android:apk-key-hash:pNiP5iKyQ8JwgGOaKA1zGPUPJIS-0H1xKCQcfIoGLck");
+        Origin https_examplecom_default = new Origin("https://example.com");
+
+        assertAll(
+                () -> assertThat(apkCertHashSha256_1).isEqualTo(apkCertHashSha256_2),
+                () -> assertThat(apkCertHashSha256_1).isNotEqualTo(apkCertHashSha1_1),
+                () -> assertThat(apkCertHashSha256_1).isNotEqualTo(https_examplecom_default),
+                () -> assertThat(apkCertHashSha256_2).isNotEqualTo(https_examplecom_default)
+        );
+    }
     @Test
     void getter_test() {
         Origin https_examplecom_default = new Origin("https://example.com");
         assertAll(
                 () -> assertThat(https_examplecom_default.getScheme()).isEqualTo("https"),
                 () -> assertThat(https_examplecom_default.getHost()).isEqualTo("example.com"),
-                () -> assertThat(https_examplecom_default.getPort()).isEqualTo(443)
+                () -> assertThat(https_examplecom_default.getPort()).isEqualTo(443),
+                () -> assertThat(https_examplecom_default.getApkSigningCertHash()).isNull(),
+                () -> assertThat(https_examplecom_default.getOriginType()).isEqualTo(Origin.OriginType.WEB)
         );
     }
 
+    @Test
+    void apk_key_hash_getter_test() {
+        final Origin apkKeyHashSha1_1 = new Origin("android:apk-key-hash:pNiP5iKyQ8JwgGOaKA1zGPUPJIS-0H1xKCQcfIoGLck");
+        assertAll(
+                () -> assertThat(apkKeyHashSha1_1.getScheme()).isEqualTo("android:apk-key-hash"),
```

Comment:
I removed the scheme check from the `Origin` class in the follow up commit because future authenticators and relying parties may support schemes other than "android". https://github.com/webauthn4j/webauthn4j/pull/354/commits/52c419a843403c8edcef0f55233b14747fa8a3d3
the scheme is still be checked by `OriginValidator` as the validator compares the origin to the expected origins

Reply:
Yes, good point. 
change set looks ok (including https://github.com/webauthn4j/webauthn4j/pull/354).

Additional  follow up, I posted https://github.com/webauthn4j/webauthn4j/pull/356 (mostly about scheme & host case insensitivity and assigning schemeSpecificPart for http/https scheme)

Reply:
Thank you for the pull-request. I merged.